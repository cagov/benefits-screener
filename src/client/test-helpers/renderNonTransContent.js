import React from "react";
import { shallow } from "enzyme";
import { I18nextProvider, Trans } from "react-i18next";
import i18n from "./i18nForTests";
import shallowUntilTarget from "./shallowUntilTarget";

function renderNonTransContent(Component, props = {}) {
  const providersWrapper = shallow(
    <I18nextProvider i18n={i18n}>
      <Component {...props} />
    </I18nextProvider>
  );

  const wrapper = shallowUntilTarget(providersWrapper, Component.name);
  // Incorrectly using renderNonTransContent to render a component that
  // contains a Trans component results in a silent failure (the rendered
  // snapshot will contain the Trans component itself instead of the rendered
  // contents). This test makes that silent failure more obvious.
  expect(wrapper.containsMatchingElement(<Trans />)).toBe(false);

  return wrapper;
}

module.exports = renderNonTransContent;
