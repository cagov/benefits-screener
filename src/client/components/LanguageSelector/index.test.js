import renderNonTransContent from "../../test-helpers/renderNonTransContent";
import Component from "./index";

describe("<LanguageSelector />", () => {
  it("renders the component", async () => {
    const wrapper = renderNonTransContent(Component, "LanguageSelector");

    expect(wrapper).toMatchSnapshot();
  });
});
