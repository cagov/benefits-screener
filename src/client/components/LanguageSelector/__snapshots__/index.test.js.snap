// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LanguageSelector /> renders the component 1`] = `
<Dropdown
  navbar={false}
>
  <DropdownToggle
    className="text-dark bg-light"
    variant="outline-secondary"
  >
    Language
  </DropdownToggle>
  <DropdownMenu
    alignRight={false}
    flip={true}
  >
    <DropdownItem
      as={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "displayName": "SafeAnchor",
          "render": [Function],
        }
      }
      disabled={false}
      onClick={[Function]}
    >
      English
    </DropdownItem>
    <DropdownItem
      as={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "displayName": "SafeAnchor",
          "render": [Function],
        }
      }
      disabled={false}
      onClick={[Function]}
    >
      Espa√±ol
    </DropdownItem>
  </DropdownMenu>
</Dropdown>
`;
